### PR TITLE
Runtest changes to use gitlog.txt

### PR DESCRIPTION
**Pull request by dodo721**
Checks the server for production before running tests

### CODE REVIEW

File: runtest.js [hunk 1]
```diff
@@ -66,5 +70,30 @@ process.env.__CONFIGURATION = JSON.stringify(config);
 // Setting real ARGV
 process.argv = argv;
 
-// Execute Jest. Specific target optional.
-shell.exec(`npm run test ${testPath} ${runInBand}`);
+const isLocal = config.site === "local";
+const infoURL = (isLocal ? "http://" : "https://") + config.site + "my.good-loop.com/build/gitlog.txt";
```
Comment (winterstein) [inline runtest.js:74]: Let's factor the my.good-loop.com bit out to a variable at the top of the file. 
We want to keep runtest.js as close as possible to the wwappbase.js/template version.

File: runtest.js [hunk 2]
```diff
@@ -66,5 +70,30 @@ process.env.__CONFIGURATION = JSON.stringify(config);
...
+
+if (!yargv.skipProdTest) {
+	// Check tests are not running on production
+	console.log("Checking gitlog for host type...");
+	fetch(infoURL, { method: 'GET', timeout:10000 })
+		.then(res => res.text())
+		.then(gitlog => {
+			const hostname = gitlog.split('\n')[0].replace(/HOST:\t/g, "");
+			// test server hostname = baker
```
Comment (winterstein) [inline runtest.js:83]: Let's factor specific setup, like "baker" to variables at the top of the file.

File: runtest.js [hunk 3]
```diff
@@ -66,5 +70,30 @@ process.env.__CONFIGURATION = JSON.stringify(config);
...
 process.argv = argv;
 
-// Execute Jest. Specific target optional.
-shell.exec(`npm run test ${testPath} ${runInBand}`);
+const isLocal = config.site === "local";
+const infoURL = (isLocal ? "http://" : "https://") + config.site + "my.good-loop.com/build/gitlog.txt";
+
+if (!yargv.skipProdTest) {
+	// Check tests are not running on production
+	console.log("Checking gitlog for host type...");
```
Comment (winterstein) [inline runtest.js:78]: Maybe an extra comment about the external assumptions - e g.

// Assume gitstamp.sh was run when the web-app was built, creating gitlog.txt

**Comment by dodo721** [comment]
Refractored to be nicer
Once this is good and merged, I could replace runtest.js on the other projects too? with care to accomodate for any specificities of those files ofc

**Comment by winterstein** [comment]
> Refractored to be nicer
> Once this is good and merged, I could replace runtest.js on the other projects too? with care to accomodate for any specificities of those files ofc

The rule is: when we change runtest.js (or one of the other "project template" files) -- we must also copy the edits into the  `wwappbase.js/template` version. So that wwappbase.js always holds the best version.

If it's a breaking change -- then you need to talk to Dan A about updating all the projects.

For a non-breaking change -- you could update some, but it's not essential.

**Pull request merged**